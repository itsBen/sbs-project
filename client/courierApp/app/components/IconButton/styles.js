import { StyleSheet } from 'react-native';

const defaultStyles = {
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'rgb(230,37,101)',
    borderWidth: 1,
    borderColor: 'rgb(230,37,101)',
    paddingVertical: 5,
    paddingHorizontal: 5,
    borderRadius: 5,
  },
  icon: {
    color: '#fff',
  },
  label: {
    color: '#fff',
    fontWeight: 'bold',
    marginTop: 5,
    fontSize: 13,
  },
};

export default {
  default: StyleSheet.create({ ...defaultStyles }),
  white: StyleSheet.create({
    container: {
      ...defaultStyles.container,
      backgroundColor: 'white',
      borderColor: '#808080',
    },
    icon: {
      ...defaultStyles.icon,
      color: '#808080',
    },
    label: {
      ...defaultStyles.label,
      color: '#808080',
    },
  }),
};
