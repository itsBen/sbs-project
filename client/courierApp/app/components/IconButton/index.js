import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import styles from './styles';

export default ({
  iconName,
  label = null,
  colorScheme = 'default',
  onPress,
}) => (
  <TouchableOpacity onPress={onPress}>
    <View
      style={
        styles[colorScheme]
          ? styles[colorScheme].container
          : styles.default.container
      }
    >
      <Icon
        name={iconName}
        size={20}
        style={
          styles[colorScheme] ? styles[colorScheme].icon : styles.default.icon
        }
      />
      {!label || (
        <Text
          style={
            styles[colorScheme]
              ? styles[colorScheme].label
              : styles.default.label
          }
        >
          {label}
        </Text>
      )}
    </View>
  </TouchableOpacity>
);

const defaultStyles = {
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'rgb(230,37,101)',
    borderWidth: 1,
    borderColor: 'rgb(230,37,101)',
    paddingVertical: 5,
    paddingHorizontal: 5,
    borderRadius: 5,
  },
  icon: {
    color: '#fff',
  },
  label: {
    color: '#fff',
    fontWeight: 'bold',
    marginTop: 5,
    fontSize: 13,
  },
};
