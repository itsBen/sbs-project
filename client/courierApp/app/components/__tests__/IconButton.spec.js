/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import IconButton from '../IconButton';

it('renders correctly', () => {
  const tree = renderer
    .create(<IconButton iconName="location-on" onPress={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
