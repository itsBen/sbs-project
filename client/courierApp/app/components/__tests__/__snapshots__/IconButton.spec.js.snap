// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(230,37,101)",
        "borderColor": "rgb(230,37,101)",
        "borderRadius": 5,
        "borderWidth": 1,
        "justifyContent": "center",
        "paddingHorizontal": 5,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "color": "#fff",
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      îƒˆ
    </Text>
  </View>
</View>
`;
